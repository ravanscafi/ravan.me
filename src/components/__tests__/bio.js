import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"

import Bio from "../bio"

beforeEach(() => {
  useStaticQuery.mockImplementationOnce(() => ({
    avatar: {
      childImageSharp: {
        fixed: {
          base64: "data:image/jpeg;base64,/9j/SHAcv//Z",
          width: 150,
          height: 150,
          src: "/static/src/profile-pic.jpg",
          srcSet: "/static/srcset/profile-pic.jpg",
        },
      },
    },
    site: {
      siteMetadata: {
        author: "Ravan Scafi",
        social: {
          twitter: "ravanscafi",
          github: "ravanscafi",
          exercism: "ravanscafi",
          linkedin: "ravanscafi",
          stackoverflow: "804741/ravan-scafi",
        },
      },
    },
  }))
})

describe("Bio", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Bio>
          <h1>John Doe</h1>
        </Bio>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
