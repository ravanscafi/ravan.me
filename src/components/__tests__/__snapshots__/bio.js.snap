// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio renders correctly 1`] = `
<div
  style={
    Object {
      "marginBottom": "4.375rem",
      "textAlign": "center",
    }
  }
>
  <div>
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "borderRadius": "100%",
          "display": "inline-block",
          "height": 150,
          "marginBottom": 0,
          "marginRight": "0.875rem",
          "overflow": "hidden",
          "position": "relative",
          "width": 150,
        }
      }
    >
      <img
        alt=""
        src="data:image/jpeg;base64,/9j/SHAcv//Z"
        style={
          Object {
            "borderRadius": "50%",
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <img
          alt="Ravan Scafi"
          height={150}
          loading="lazy"
          onLoad={[Function]}
          src="/static/src/profile-pic.jpg"
          srcSet="/static/srcset/profile-pic.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
          width={150}
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><img loading=\\"lazy\\" width=\\"150\\" height=\\"150\\" srcset=\\"/static/srcset/profile-pic.jpg\\" src=\\"/static/src/profile-pic.jpg\\" alt=\\"Ravan Scafi\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "1 1 auto",
      }
    }
  >
    <h1>
      John Doe
    </h1>
    I am a web developer living in SÃ£o Paulo - Brazil.
    <div
      style={
        Object {
          "marginTop": "1.75rem",
        }
      }
    >
      <a
        className="social-icon twitter"
        href="https://twitter.com/ravanscafi"
        rel="nofollow"
        title="Twitter"
      >
        <svg
          className="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
          />
        </svg>
      </a>
      <a
        className="social-icon github"
        href="https://github.com/ravanscafi"
        rel="nofollow"
        title="GitHub"
      >
        <svg
          className="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"
          />
        </svg>
      </a>
      <a
        className="social-icon stackoverflow"
        href="https://stackoverflow.com/users/804741/ravan-scafi"
        rel="nofollow"
        title="Stack Overflow"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 384 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M290.7 311L95 269.7 86.8 309l195.7 41zm51-87L188.2 95.7l-25.5 30.8 153.5 128.3zm-31.2 39.7L129.2 179l-16.7 36.5L293.7 300zM262 32l-32 24 119.3 160.3 32-24zm20.5 328h-200v39.7h200zm39.7 80H42.7V320h-40v160h359.5V320h-40z"
          />
        </svg>
      </a>
      <a
        className="social-icon linkedin"
        href="https://www.linkedin.com/in/ravanscafi"
        rel="nofollow"
        title="Linkedin"
      >
        <svg
          className="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 8a6 6 0 0 1 6 6v7h-4v-7a2 2 0 0 0-2-2 2 2 0 0 0-2 2v7h-4v-7a6 6 0 0 1 6-6z"
          />
          <rect
            height="12"
            width="4"
            x="2"
            y="9"
          />
          <circle
            cx="4"
            cy="4"
            r="2"
          />
        </svg>
      </a>
    </div>
  </div>
</div>
`;
